### PR TITLE
CLN: remove keep_null_freq in ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
This cleanup is viable now that integer-addition has been disallowed for dt64/td64.

### CODE REVIEW

File: pandas/tests/arithmetic/test_timedelta64.py [hunk 1]
```diff
@@ -1046,6 +1046,58 @@ def test_td64arr_add_sub_numeric_arr_invalid(self, box_with_array, vec, dtype):
...
+        with pytest.raises(TypeError, match=msg):
+            rng + one
+        with pytest.raises(TypeError, match=msg):
+            rng += one
+        with pytest.raises(TypeError, match=msg):
+            rng - one
+        with pytest.raises(TypeError, match=msg):
+            rng -= one
+
+    # TODO: box and de-duplicate
```
Comment (gfyoung) [inline pandas/tests/arithmetic/test_timedelta64.py]: I don't get the first part of this comment.

File: pandas/tests/arithmetic/test_timedelta64.py [hunk 2]
```diff
@@ -1046,6 +1046,58 @@ def test_td64arr_add_sub_numeric_arr_invalid(self, box_with_array, vec, dtype):
         with pytest.raises(TypeError):
             vector - tdser
 
+    # TODO: box and de-duplicate
```
Comment (gfyoung) [inline pandas/tests/arithmetic/test_timedelta64.py]:
```suggestion
    # TODO: parameterize over box and de-duplicate
```

Is that what you mean?

File: pandas/tests/arithmetic/test_timedelta64.py [hunk 3]
```diff
@@ -1046,6 +1046,58 @@ def test_td64arr_add_sub_numeric_arr_invalid(self, box_with_array, vec, dtype):
...
+        with pytest.raises(TypeError, match=msg):
+            other + rng
+
+        with pytest.raises(TypeError, match=msg):
+            rng - other
+
+        with pytest.raises(TypeError, match=msg):
+            other - rng
+
+    # TODO: box and de-duplicate
```
Comment (gfyoung) [inline pandas/tests/arithmetic/test_timedelta64.py]: Same as [above](https://github.com/pandas-dev/pandas/pull/30221#discussion_r357004201)

**Review by jreback — APPROVED** [review]
lgtm. some comments by @gfyoung

**Comment by jbrockmendel** [comment]
updated+green

**Comment by jreback** [comment]
thanks

**Pull request merged**